docs/source/tests/pytest.rst

```diff
@@ -0,0 +1,124 @@
+In order to test files, classes or methods in the current directory and subdirectories there are some helpful `commands <https://docs.pytest.org/en/6.2.x/usage.html#calling-pytest-through-python-m-pytest>`_ to execute from the terminal:
+
+.. code-block:: shell
+
+    $ pytest # to run all tests
+
+.. code-block:: shell
+
+    $ pytest <directory>/ # to run all tests in the <directory> directory
+
+.. code-block:: shell
+
+    $ pytest <filename>.py # to run tests in the <filename> file
+
+.. code-block:: shell
+
+    $ pytest -m <name> # to run all tests with the @pytest.mark.<name> decorator (see below)
+    $ pytest -m "not <name>" # to run all tests which do not have the @pytest.mark.<name> decorator (see below)
+
+.. code-block:: shell
+
+    $ pytest -k "<string1> and not <string2>" # to run all tests which contain the <string1> and not the <string2> expression
+
+.. code-block:: shell
+
+    $ pytest <filename>.py::<methode_name> # to run a specific test (<method_name>) within a module (<filename>)
+
+# Should I exclude this examples and just reference the part of the documentation of pytest which contains all these commands
+
+In order to understand the test report provided by pytest in detail this `link <https://docs.pytest.org/en/latest/how-to/output.html>`__ is recommended.
+
+
+Useful features
+===============
+
+Pytest provides multiple features that are very useful to simplify the testing procedure and keep your tests organized and structured.
+
+Fixtures
+---------
+Most of the tests depend on some sort of input. With `fixtures <https://docs.pytest.org/en/6.2.x/fixture.html>`_ pytest provides a feature with which data, test doubles or some system state can be created. Fixtures are reusable and can be used for multiple tests. In order to create a fixture you have to build a function that returns the data or system state that is needed for your testing. To do that just decorate this function with ``@pytest.fixture``. The function name can now get passed to a testing method as an argument. Here you can see an example of how fixtures can be implemented:
+
+.. code-block:: python
+
+    ''' tests.test_core.test_xy '''
+    import pytest
+    import numpy as np
+
+    from floodlight.core.xy import XY
+
+    # creation of the fixture
+    @pytest.fixture()
+    def example_xy_data_pos_int() -> np.ndarray:
+        positions = np.array([[1, 2, 3, 4], [5, 6, 7, 8]])
+        return positions
+
+    # testing a function with the fixture being passed as an argument
+    def test_x_pos_int(example_xy_data_pos_int: np.ndarray) -> None:
+        # Arrange
+        data = XY(example_xy_data_pos_int)
+
+        # Act
+        x_position = data.x
+
+        # Assert
+        assert np.array_equal(x_position, np.array([[1, 3], [5, 7]]))
+
+Fixtures are a quite powerful tool since they are modular and can also request other fixtures.
```

Comment:
Maybe one more sentence here for how we use them: to get minimal examples of e.g. data-level objects such as XY, Events, or Code. Of course, they look different depending on the method tested

Reply:
Added: In a nutshell they can be understood as minimal examples of e.g. data-level objects such as XY, Events, or Code. But compared to the normal objects, they are much clearer and are still able to test the full functionality of the methods. Of course, they look different depending on the method tested.